src/libpmemkv.cc

```diff
@@ -30,22 +30,35 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <rapidjson/document.h>
+#if defined(ENABLE_VSMAP_ENGINE) || defined(ENABLE_VCMAP_ENGINE) || defined(ENABLE_CMAP_ENGINE) ||\
+    defined(ENABLE_STREE_ENGINE) || defined(ENABLE_TREE3_ENGINE)
 #include <libpmemobj++/make_persistent.hpp>
```

Comment:
These includes shouldn't be needed here. Each engine that requires them should include them.